test/functional/test_files.py

```diff
@@ -126,11 +135,57 @@ def test_only_hash_file(client):
 	assert res["Hash"] not in client.pin.ls(type="recursive")
 	assert res["Hash"] not in list(map(lambda i: i["Ref"], client.unstable.refs.local()))
 
-
 def test_add_multiple_from_list(client, cleanup_pins):
 	res = client.add([FAKE_FILE1_PATH, FAKE_FILE2_PATH])
 	assert FAKE_FILES_HASH == res
 
+def test_add_with_raw_leaves(client):
```

Comment:
Please add the `cleanup_pins` fixture to all of your tests, like so:

```py
def test_add_with_raw_leaves(client, cleanup_pins):
	# Body
```

This way the pins implicitly added will be removed again after the test.

Reply:
Added fixture to non-error test cases.